addon/modifiers/did-intersect.js

```diff
@@ -1,44 +1,42 @@
 import Modifier from 'ember-modifier';
+import { inject as service } from '@ember/service';
 
 export default class DidIntersectModifier extends Modifier {
-  // Public API
-  handler = null;
-  options = {};
+  @service observerManager;
 
-  // Private API
-  observer = null;
-
-  observe() {
+  observe(options) {
     if ('IntersectionObserver' in window) {
-      this.observer = new IntersectionObserver((entries, observer) => {
-        this.handler(entries[0], observer);
-      }, this.options);
-
-      this.observer.observe(this.element);
+      this.observerManager.observe(this.element, options);
     }
   }
 
   unobserve() {
-    if (this.observer) {
-      this.observer.disconnect();
-    }
+    this.observerManager.unobserve(this.element);
   }
 
   didUpdateArguments() {
     this.unobserve();
   }
 
   didReceiveArguments() {
+    debugger;
     let [handler, options] = this.args.positional;
 
-    // Save arguments for when we need them
-    this.handler = handler;
-    this.options = options || this.options;
+    const { onEnter, onExit } = handler;
```

Comment:
I don't think we want `onEnter` and `onExit` to come from a `handler` object. Ideally, we would ditch positional arguments at this point as a breaking change and go for named arguments instead. That is:

```
let { onEnter, onExit, options } = this.args.named;
```

And then the modifier would be invoked like so:

```
{{did-intersect onEnter=this.onEnter onExit=this.onExit options=options}}
```
```